modules/jsdom/src/polyfills/events/mouse.ts

```diff
@@ -37,19 +37,34 @@ export function mouseEvents(window: DOMWindow) {
 
 function buttonUpdates(window: DOMWindow) {
   return windowCallbackAsObservable(glfw.setMouseButtonCallback, window)
-    .pipe(map(([, ...rest]) => GLFWMouseEvent.fromMouseButton(window, ...rest)))
+    .pipe(flatMap(function*([, ...rest]) {
+      const mouseEvt = GLFWMouseEvent.fromMouseButton(window, ...rest);
+      yield mouseEvt;
+      yield Object.assign(GLFWMouseEvent.fromMouseButton(window, ...rest),
+                          {type: `pointer${mouseEvt.type.slice(5)}`});
+    }))
     .pipe(publish(), refCount());
 }
 
 function positionUpdates(window: DOMWindow) {
   return windowCallbackAsObservable(glfw.setCursorPosCallback, window)
-    .pipe(map(([, ...rest]) => GLFWMouseEvent.fromMouseMove(window, ...rest)))
+    .pipe(flatMap(function*([, ...rest]) {
+      const mouseEvt = GLFWMouseEvent.fromMouseMove(window, ...rest);
+      yield mouseEvt;
+      yield Object.assign(GLFWMouseEvent.fromMouseMove(window, ...rest),
+                          {type: `pointer${mouseEvt.type.slice(5)}`});
```

Comment:
Similar to `function*`, what is `pointer$` here?

Reply:
this is making event type from the current `mouse<blah>` to `pointer<blah>`.

Reply:
Oh duh it is just a `python f string`.